### ISSUE TITLE
Example app build issues

### ISSUE THREAD

**Issue by pmagnuson** [cph-cachet/research.package#86]
after `git clone git@github.com:cph-cachet/research.package.git`

removed `flutter_cupertino_localizations: ^1.0.1` from `pubspec.yaml` to build.

Prior to remove, encountered these errors:

`flutter pub get` had these issues:
```
[example] flutter pub get
Resolving dependencies...
The current Dart SDK version is 3.0.1.

Because research_package_demo_app depends on flutter_cupertino_localizations >=1.0.1 which doesn't support null safety, version solving failed.

The lower bound of "sdk: '>=2.0.0-dev.68.0 <3.0.0'" must be 2.12.0 or higher to enable null safety.
For details, see https://dart.dev/null-safety
exit code 1
```

`flutter doctor -v`
```

[✓] Flutter (Channel stable, 3.10.1, on macOS 13.0.1 22A400 darwin-arm64, locale en-US)
    • Flutter version 3.10.1 on channel stable at /Users/phil/bin/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d3d8effc68 (5 weeks ago), 2023-05-16 17:59:05 -0700
    • Engine revision b4fb11214d
    • Dart version 3.0.1
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/phil/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • ANDROID_HOME = /Users/phil/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.79.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (5 available)
    • Pixel 5a (mobile)  • 19211JECB03038                       • android-arm64  • Android 13 (API 33)
    • iPhone12 (mobile)  • 00008101-000E706C3641001E            • ios            • iOS 16.3.1 20D67
    • iPhone 14 (mobile) • 0A042922-6107-47BB-9F56-05D84CBFF2F2 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 13.0.1 22A400 darwin-arm64
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 114.0.5735.133
    ! Error: Phil’s Apple Watch has not finished loading development services. Please select a different device, or wait for the device to load
      development services and try again. (code 8)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

**Comment by jjspoc** [comment]
Lucky, you. I unfortunately can´t get this or any other example (e.g the cognition/concenting/survey)  to run (Win10 environment, VSC, flutter and dart are up-to-date, Android Studio 2022). 


pub_spec.yaml:

```
....snip....
  cupertino_icons: ^1.0.5
#  flutter_cupertino_localizations: ^1.0.1
  flutter_localizations:
    sdk: flutter
....snip....
```

This got rid of the nullable errors; the   flutter_localizations dependency is still required. Commenting out throws many more error concerning localization.

What I do see now relates to something else. As you may have guessed I´m not a developer and are poking somewhat in the dark. 

```
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/03 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/03 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: unexpected element (uri:"", local:"base-extension"). Expected elements are <{}codename>,<{}layoutlib>,<{}api-level>
Invalid depfile: C:\Users\Public\Documents\Code\Copenhagen\research.package\example\.dart_tool\flutter_build\b924db852b802065d95ccb24f28c825f\kernel_snapshot.d
Invalid depfile: C:\Users\Public\Documents\Code\Copenhagen\research.package\example\.dart_tool\flutter_build\b924db852b802065d95ccb24f28c825f\kernel_snapshot.d
../lib/src/ui/RPUITask.dart:506:18: Error: The 'records' language feature is disabled for this library.
Try removing the package language version or setting the language version to 3.0 or higher.
      case const (List<RPChoice>,):
                 ^
Target kernel_snapshot failed: Exception 
```
The example does not run for windows or chrome either.
I cannot exclude something is wrong with my setup and do appreciate any help or pointer to help,

Best,

Jörg

**Comment by jjspoc** [comment]
to whom it may concern...

after a long time fiddling around with pubspec yamls and folder structures I managed to run cognition package and the research_package example apps (so far on the android phone emulator, windows and web). In essence it was a mixture of:

- wrong gradle versions specified in  ../android/gradle/wrapper/gradle-wrapper.properties (check compatibility matrix at https://docs.gradle.org/current/userguide/compatibility.html to match against your jdk version. I'm now running JDK 19 and gradle 7.6
- pubspec needs a higher version for the cupertino_icons: '^1.0.5'
- cleaning some code issues with 'dart fix --dry-run' and a subsequent 'dart fix --apply'
- icons and images need to huddle in the right asset folder 
- my lack of experience and skills

I got to run it successfully once I extracted the example folder into a newly created project folder, e.g. 'flutter create cognitionapp' or demoapp, copying the example and asset files and adjusting the gradle versions.

@Cachet, great stuff your are developing!!
cheers,

Jörg
